---
title: "Environmental determinants differentiate metabolically active soil microorganisms from the total community: the influence of pH and soil moisture"
output: html_document
---

#### **Authors:** [Karl J. Romanowicz](mailto:kjromano@umich.edu), Zachary B. Freedman, Rima A. Upchurch, William A. Argiroff, Donald R. Zak
#### **GitHub Repository:** https://github.com/kromanowicz/MI-Gradient-Romanowicz-2016
#### **In Review for:** *FEMS Microbiology Ecology*
#### **Submitted for Review:** March 22, 2016

###Set the Global Knitr Options
```{r setup, include=TRUE}
# Set the global options
knitr::opts_chunk$set(echo = TRUE, eval = TRUE, fig.width = 6, fig.height = 5, fig.align = 'center', warning = FALSE, message = FALSE)

# Set the seed
set.seed(183409)
```

###Load in R-Packages and Set the Working Environment
```{r}
# Load packages
necessary_packages <- c("data.table","vegan","ggplot2","grid","gridExtra","devtools","plyr","pander","reshape2","scales")

# Install the necessary packages
packages <- lapply(necessary_packages, library, character.only = TRUE)

# Set the working directory
setwd("~/Desktop/Stats")
```

###Set Table Options
```{r}
#Set pander (table) output functions
panderOptions('table.split.table', Inf)
panderOptions('table.style', 'rmarkdown')

# Do not show scientific notation
options(scipen = 999, digits = 7)
```

###Import MOTHUR data
```{r MOTHUR_.shared}
# Load in the shared file from the MOTHUR analysis
bacteria<-read.table("c.g.DNA.16S.combined.trim.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.03.subsample.0.03.shared", header=TRUE)

fungi<-read.table("c.g.DNA.28S.combined.trim.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.01.subsample.0.01.shared", header=TRUE)

# Remove the first 3 columns from each .shared file
bacteria<-bacteria[,-c(1:3)]

fungi<-fungi[,-c(1:3)]

# Add back in a new column with Community and Site combined
bacteria<-data.frame(Treatment = c("Total_A_1","Total_A_2","Total_A_3","Total_B_1","Total_B_2","Total_B_3","Total_C_1","Total_C_2","Total_C_3","Total_D_1","Total_D_2","Total_D_3","Active_A_1","Active_A_2","Active_A_3","Active_B_1","Active_B_2","Active_B_3","Active_C_1","Active_C_2","Active_C_3","Active_D_1","Active_D_2","Active_D_3"), bacteria)

fungi<-data.frame(Treatment = c("Total_A_1","Total_A_2","Total_A_3","Total_B_1","Total_B_2","Total_B_3","Total_C_1","Total_C_2","Total_C_3","Total_D_1","Total_D_2","Total_D_3","Active_A_1","Active_A_2","Active_A_3","Active_B_1","Active_B_2","Active_B_3","Active_C_1","Active_C_2","Active_C_3","Active_D_1","Active_D_2","Active_D_3"), fungi)

# Convert the values in Column 1 ("Treatment") into row names
rownames(bacteria)<-bacteria[,1]
bacteria<-bacteria[,-1]

rownames(fungi)<-fungi[,1]
fungi<-fungi[,-1]
```

```{r MOTHUR_groups.summary}
# Call in the MOTHUR "groups.summary" file for Bacterial Alpha Diversity metrics
bact.alpha <- read.table("c.g.DNA.16S.combined.trim.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.03.subsample.0.03.groups.summary", header=TRUE)

# Convert to data.table format
bact.alpha2 <- data.table(bact.alpha)

# Add new columns for "Community" and "Site" to the end of the data table
bact.alpha2$Site <- c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")
bact.alpha2$Community <- c("Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total")

# Separate Total and Active Communities
bact.total.alpha<-bact.alpha2[c(13:24),]
bact.active.alpha<-bact.alpha2[c(1:12),]

# Call in the MOTHUR "groups.summary" file for Fungal Alpha Diversity metrics
fungi.alpha <- read.table("c.g.DNA.28S.combined.trim.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.01.subsample.0.01.groups.summary", header=TRUE)

# Convert to data.table format
fungi.alpha2 <- data.table(fungi.alpha)

# Add new columns for "Community" and "Site" to the end of the data table
fungi.alpha2$Site <- c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")
fungi.alpha2$Community <- c("Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total")

# Separate Total and Active Communities
fungi.total.alpha<-fungi.alpha2[c(13:24),]
fungi.active.alpha<-fungi.alpha2[c(1:12),]
```

###Import Environmental Meta-data
```{r Env}
# Call in "env.txt" file with meta-data
env<-read.table("env.txt",header=TRUE)

new.env<-env

# Add new columns for "Community" and "Site" to the end of the data table
new.env$Site<-c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")
new.env$Community<-c("Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active")
```

###Recreating Table 1 - Environmental Metadata
```{r Table_1}
# Remove Community column from env file
new.env2<-new.env[,-11]

# Remove rows 13:24 (redundant env between active and total)
new.env2<-new.env2[-c(13:24), ]

# Calculate mean of each column by Site
table1.mean<-ddply(new.env2, .(Site), colwise(mean))

# Calculate sd of each column by Site
table1.sd<-ddply(new.env2, .(Site), colwise(sd))

### Transpose data frame

# Remember Sites as non-numeric values
table1.mean.sites<-table1.mean$Site
table1.sd.sites<-table1.sd$Site

# Transpose all but first column (Site)
table1.mean<-as.data.frame(t(table1.mean[,-1]))
colnames(table1.mean)<-table1.mean.sites
table1.mean<-format(round(table1.mean,1))

table1.sd<-as.data.frame(t(table1.sd[,-1]))
colnames(table1.sd)<-table1.sd.sites
table1.sd<-format(round(table1.sd,1))

# Combine mean and sd into single column with ± divider
table1.combine<-as.data.frame(do.call(cbind, lapply(1:ncol(table1.mean), function(i) paste0(table1.mean[ , i], " ±", table1.sd[ , i]))))

# Rename rows in the data frame to environmental variables
row.names(table1.combine)<-c("Moisture","pH","Aryl","Alkyl","O-alkyl","Carboxyl","C","N","C:N")

# Rename columns to Sites
colnames(table1.combine)<-c("Site A","Site B","Site C","Site D")

# Set pander options (table formatting)
set.alignment('right',row.names = 'left')
pander(table1.combine,caption = "Table 1. Physical, chemical, and biogeochemical soil characteristics for four forest stands in upper and lower Michigan, USA that span the north-south geographic range of northern hardwoods. Mean values are presented.")
```

###Statistics on Environmental Differences Across Sites
```{r Env_Stats}
#Create matrix of numeric columns
env.bind.mat<-as.matrix(cbind(new.env2[,1:9]))

# MANOVA
env.manova<-manova(env.bind.mat~Site, data=new.env2)
env.man.sum<-summary.aov(env.manova)
env.man.sum

# Run individual ANOVA on those env factors with significant Site differences to get at Post-hoc analysis

# Moisture ANOVA with Post-Hoc for Site Differences
env.moisture<-aov(Moisture~Site, data=new.env2)
TukeyHSD(env.moisture)

# pH ANOVA with Post-Hoc for Site Differences
env.ph<-aov(pH~Site, data=new.env2)
TukeyHSD(env.ph)

# Aryl ANOVA with Post-Hoc for Site Differences
env.aryl<-aov(Aryl~Site, data=new.env2)
TukeyHSD(env.aryl)

# Carboxyl ANOVA with Post-Hoc for Site Differences
env.carboxyl<-aov(Carboxyl~Site, data=new.env2)
TukeyHSD(env.carboxyl)

# N ANOVA with Post-Hoc for Site Differences
env.n<-aov(N~Site, data=new.env2)
TukeyHSD(env.n)

# CN ANOVA with Post-Hoc for Site Differences
env.cn<-aov(CN~Site, data=new.env2)
TukeyHSD(env.cn)
```

###Alpha Diversity - Bacterial Chao1 Richness
```{r Alpha_Bacteria}
# Bacterial alpha diversity for Site x Community Interactions - Chao1 Richness
chao.bact<-aov(chao~Site*Community, data=bact.alpha2)
summary.aov(chao.bact)

#Tukey's HSD Post-hoc analysis
TukeyHSD(chao.bact)
```

###Recreating Table 2. Alpha Diversity Bacterial Chao1 Richness
```{r Table_2_Bacteria}
bact.alpha3<-bact.alpha

# Add new columns for "Community" and "Site" to the end of the data table
bact.alpha3$Site <- c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")

# Add new column for Site and Community combined
bact.alpha3$ID<-c("Active_A_1","Active_A_2","Active_A_3","Active_B_1","Active_B_2","Active_B_3","Active_C_1","Active_C_2","Active_C_3","Active_D_1","Active_D_2","Active_D_3","Total_A_1","Total_A_2","Total_A_3","Total_B_1","Total_B_2","Total_B_3","Total_C_1","Total_C_2","Total_C_3","Total_D_1","Total_D_2","Total_D_3")

# Make ID as rownames
row.names(bact.alpha3)<-bact.alpha2$ID

# Keep only columns "chao", "Site", and "Community"
bact.alpha3<-bact.alpha3[,-c(1:3,5:6,8)]

# Separate Active and Total Communities
bact.alpha2.total<-bact.alpha3[13:24,]
bact.alpha2.active<-bact.alpha3[1:12,]

# Calculate mean of each column by Site
table2.bact.total.mean<-ddply(bact.alpha2.total, .(Site), colwise(mean))
table2.bact.active.mean<-ddply(bact.alpha2.active, .(Site), colwise(mean))

# Calculate sd of each column by Site
table2.bact.total.sd<-ddply(bact.alpha2.total, .(Site), colwise(sd))
table2.bact.active.sd<-ddply(bact.alpha2.active, .(Site), colwise(sd))

### Transpose data frame

# Remember Sites as non-numeric values
table2.bact.total.mean.sites<-table2.bact.total.mean$Site
table2.bact.active.mean.sites<-table2.bact.active.mean$Site

table2.bact.total.sd.sites<-table2.bact.total.sd$Site
table2.bact.active.sd.sites<-table2.bact.active.sd$Site

# Transpose all but first column (Site)
table2.bact.total.mean<-as.data.frame(t(table2.bact.total.mean[,-1]))
colnames(table2.bact.total.mean)<-table2.bact.total.mean.sites
table2.bact.total.mean<-format(round(table2.bact.total.mean,0))

table2.bact.active.mean<-as.data.frame(t(table2.bact.active.mean[,-1]))
colnames(table2.bact.active.mean)<-table2.bact.active.mean.sites
table2.bact.active.mean<-format(round(table2.bact.active.mean,0))

table2.bact.total.sd<-as.data.frame(t(table2.bact.total.sd[,-1]))
colnames(table2.bact.total.sd)<-table2.bact.total.sd.sites
table2.bact.total.sd<-format(round(table2.bact.total.sd,0))

table2.bact.active.sd<-as.data.frame(t(table2.bact.active.sd[,-1]))
colnames(table2.bact.active.sd)<-table2.bact.active.sd.sites
table2.bact.active.sd<-format(round(table2.bact.active.sd,0))

# Combine mean and sd into single column with ± divider
table2.bact.total.combine<-as.data.frame(do.call(cbind,lapply(1:ncol(table2.bact.total.mean),function(i) paste0(table2.bact.total.mean[ , i], " ± ", table2.bact.total.sd[ , i]))))

table2.bact.active.combine<-as.data.frame(do.call(cbind,lapply(1:ncol(table2.bact.active.mean),function(i) paste0(table2.bact.active.mean[ , i], " ± ", table2.bact.active.sd[ , i]))))

# Transpose tables
table2.bact.total.combine<-as.data.frame(t(table2.bact.total.combine))
table2.bact.active.combine<-as.data.frame(t(table2.bact.active.combine))

# Add new column with values from column V1
colnames(table2.bact.total.combine)<-"Total"
colnames(table2.bact.active.combine)<-"Active"

# Add Site column
table2.bact.total.combine$Site<-c("Site A","Site B","Site C","Site D")
table2.bact.active.combine$Site<-c("Site A","Site B","Site C","Site D")

# Merge Total and Active together
table2.bact.alpha<-merge(table2.bact.total.combine,table2.bact.active.combine,by="Site",all=TRUE)

# Create Mean difference column
table2.bact.alpha$Difference<-as.numeric(table2.bact.active.mean[1,])-as.numeric(table2.bact.total.mean[1,])

# Set Site as rownames
row.names(table2.bact.alpha)<-table2.bact.alpha$Site

# Remove Site column
table2.bact.alpha<-table2.bact.alpha[2:4]

# Rename rows in the data frame to Sites
row.names(table2.bact.alpha)<-c("Site A","Site B","Site C","Site D")

# Rename columns
colnames(table2.bact.alpha)<-c("Total","Active","% Difference from Total")

# Set pander options (table formatting)
set.alignment('right',row.names = 'left')
pander(table2.bact.alpha,caption = "Table 2. Alpha diversity Chao1 Richness (mean ± standard deviation) for bacterial total and active communities with mean difference from total community.")
```

###Alpha Diversity - Fungal Chao1 Richness
```{r Alpha_Fungi}
# Fungal alpha diversity - Chao1 Richness
chao.fungi<-aov(chao~Site*Community, data=fungi.alpha2)
summary.aov(chao.fungi)

# Tukey's HSD post-hoc analysis
TukeyHSD(chao.fungi)
```

###Recreating Table 2. Alpha Diversity Fungal Chao1 Richness
```{r Table_2_Fungi}
fungi.alpha3<-fungi.alpha

# Add new columns for "Community" and "Site" to the end of the data table
fungi.alpha3$Site <- c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")

# Add new column for Site and Community combined
fungi.alpha3$ID<-c("Active_A_1","Active_A_2","Active_A_3","Active_B_1","Active_B_2","Active_B_3","Active_C_1","Active_C_2","Active_C_3","Active_D_1","Active_D_2","Active_D_3","Total_A_1","Total_A_2","Total_A_3","Total_B_1","Total_B_2","Total_B_3","Total_C_1","Total_C_2","Total_C_3","Total_D_1","Total_D_2","Total_D_3")

# Make ID as rownames
row.names(fungi.alpha3)<-fungi.alpha3$ID

# Keep only columns "chao", "Site", and "Community"
fungi.alpha3<-fungi.alpha3[,-c(1:3,5:6,8)]

# Separate Active and Total Communities
fungi.alpha2.total<-fungi.alpha3[13:24,]
fungi.alpha2.active<-fungi.alpha3[1:12,]

# Calculate mean of each column by Site
table2.fungi.total.mean<-ddply(fungi.alpha2.total, .(Site), colwise(mean))
table2.fungi.active.mean<-ddply(fungi.alpha2.active, .(Site), colwise(mean))

# Calculate sd of each column by Site
table2.fungi.total.sd<-ddply(fungi.alpha2.total, .(Site), colwise(sd))
table2.fungi.active.sd<-ddply(fungi.alpha2.active, .(Site), colwise(sd))

### Transpose data frame

# Remember Sites as non-numeric values
table2.fungi.total.mean.sites<-table2.fungi.total.mean$Site
table2.fungi.active.mean.sites<-table2.fungi.active.mean$Site

table2.fungi.total.sd.sites<-table2.fungi.total.sd$Site
table2.fungi.active.sd.sites<-table2.fungi.active.sd$Site

# Transpose all but first column (Site)
table2.fungi.total.mean<-as.data.frame(t(table2.fungi.total.mean[,-1]))
colnames(table2.fungi.total.mean)<-table2.fungi.total.mean.sites
table2.fungi.total.mean<-format(round(table2.fungi.total.mean,0))

table2.fungi.active.mean<-as.data.frame(t(table2.fungi.active.mean[,-1]))
colnames(table2.fungi.active.mean)<-table2.fungi.active.mean.sites
table2.fungi.active.mean<-format(round(table2.fungi.active.mean,0))

table2.fungi.total.sd<-as.data.frame(t(table2.fungi.total.sd[,-1]))
colnames(table2.fungi.total.sd)<-table2.fungi.total.sd.sites
table2.fungi.total.sd<-format(round(table2.fungi.total.sd,0))

table2.fungi.active.sd<-as.data.frame(t(table2.fungi.active.sd[,-1]))
colnames(table2.fungi.active.sd)<-table2.fungi.active.sd.sites
table2.fungi.active.sd<-format(round(table2.fungi.active.sd,0))

# Combine mean and sd into single column with ± divider
table2.fungi.total.combine<-as.data.frame(do.call(cbind,lapply(1:ncol(table2.fungi.total.mean),function(i) paste0(table2.fungi.total.mean[ , i], " ± ", table2.fungi.total.sd[ , i]))))

table2.fungi.active.combine<-as.data.frame(do.call(cbind,lapply(1:ncol(table2.fungi.active.mean),function(i) paste0(table2.fungi.active.mean[ , i], " ± ", table2.fungi.active.sd[ , i]))))

# Transpose tables
table2.fungi.total.combine<-as.data.frame(t(table2.fungi.total.combine))
table2.fungi.active.combine<-as.data.frame(t(table2.fungi.active.combine))

# Add new column with values from column V1
colnames(table2.fungi.total.combine)<-"Total"
colnames(table2.fungi.active.combine)<-"Active"

# Add Site column
table2.fungi.total.combine$Site<-c("Site A","Site B","Site C","Site D")
table2.fungi.active.combine$Site<-c("Site A","Site B","Site C","Site D")

# Merge Total and Active together
table2.fungi.alpha<-merge(table2.fungi.total.combine,table2.fungi.active.combine,by="Site",all=TRUE)

# Create Mean difference column
table2.fungi.alpha$Difference<-as.numeric(table2.fungi.active.mean[1,])-as.numeric(table2.fungi.total.mean[1,])

# Set Site as rownames
row.names(table2.fungi.alpha)<-table2.fungi.alpha$Site

# Remove Site column
table2.fungi.alpha<-table2.fungi.alpha[2:4]

# Rename rows in the data frame to Sites
row.names(table2.fungi.alpha)<-c("Site A","Site B","Site C","Site D")

# Rename columns
colnames(table2.fungi.alpha)<-c("Total","Active","% Difference from Total")

# Set pander options (table formatting)
set.alignment('right',row.names = 'left')
pander(table2.fungi.alpha,caption = "Table 2. Alpha diversity Chao1 Richness (mean ± standard deviation) for fungi total and active communities with mean difference from total community.")
```

###Beta Diversity - PERMANOVA for Site and Community Influences

####BACTERIA
```{r Beta_Bacteria_Qualitative}
# Bacteria - Bray-Curtis Dissimilarity (OTU relative abundance)
bact.bc.dist <- vegdist(bacteria, method = "bray", binary = FALSE)

# PERMANOVA - Bacteria - Bray-Curtis
adonis.bc.bact<-adonis(bact.bc.dist~Site*Community, data=new.env)
adonis.bc.bact
```

####FUNGI
```{r Beta_Fungi_Qualitative}
# Fungi - Bray-Curtis Dissimilarity (OTU relative abundance)
fungi.bc.dist <- vegdist(fungi, method = "bray", binary = FALSE)

# PERMANOVA - Fungi - Bray-Curtis
adonis.bc.fungi<-adonis(fungi.bc.dist~Site*Community, data=new.env)
adonis.bc.fungi
```

###Beta Diversity - PERMANOVA for Environmental Influences

####Transform environmental data
```{r Env_Z_Score}
#Z-score transform environmental data
new.env.qual<-new.env
new.env.qual<-new.env.qual[1:9]
new.env.qual.z<-scale(new.env.qual)
new.env.qual.z<-as.data.frame(new.env.qual.z)

# Test for collinearity of environmental data (|r| > 0.80)
env.corr<-cor(new.env.qual.z)
env.corr
```

####BACTERIA
```{r Beta_Bacteria_Env}
#Bacteria - Moisture
adonis.bc.bact.moisture.z<-adonis(bact.bc.dist~Moisture, data=new.env.qual.z)

#Bacteria - pH
adonis.bc.bact.ph.z<-adonis(bact.bc.dist~pH, data=new.env.qual.z)

#Bacteria - Alkyl
adonis.bc.bact.alkyl.z<-adonis(bact.bc.dist~Alkyl, data=new.env.qual.z)

#Bacteria - O-alkyl
adonis.bc.bact.oalkyl.z<-adonis(bact.bc.dist~O_alkyl, data=new.env.qual.z)

#Bacteria - Carboxyl
adonis.bc.bact.carboxyl.z<-adonis(bact.bc.dist~Carboxyl, data=new.env.qual.z)

#Bacteria - C
adonis.bc.bact.c.z<-adonis(bact.bc.dist~C, data=new.env.qual.z)

#Bacteria - N
adonis.bc.bact.n.z<-adonis(bact.bc.dist~N, data=new.env.qual.z)

###------------Extracting R2 and P-Values

###Bacteria-Moisture
# Extract r2 value
bact.moisture.r<-(adonis.bc.bact.moisture.z)[[1]][["R2"]][[1]]
bact.moisture.r.df<-as.data.frame(bact.moisture.r)
# Extract p-values
bact.moisture.p<-(adonis.bc.bact.moisture.z)[[1]][["Pr(>F)"]][[1]]
bact.moisture.p.df<-as.data.frame(bact.moisture.p)

###Bacteria-pH
# Extract r2 value
bact.ph.r<-(adonis.bc.bact.ph.z)[[1]][["R2"]][[1]]
bact.ph.r.df<-as.data.frame(bact.ph.r)
# Extract p-values
bact.ph.p<-(adonis.bc.bact.ph.z)[[1]][["Pr(>F)"]][[1]]
bact.ph.p.df<-as.data.frame(bact.ph.p)

###Bacteria-Alkyl
# Extract r2 value
bact.alkyl.r<-(adonis.bc.bact.alkyl.z)[[1]][["R2"]][[1]]
bact.alkyl.r.df<-as.data.frame(bact.alkyl.r)
# Extract p-values
bact.alkyl.p<-(adonis.bc.bact.alkyl.z)[[1]][["Pr(>F)"]][[1]]
bact.alkyl.p.df<-as.data.frame(bact.alkyl.p)

###Bacteria-O_alkyl
# Extract r2 value
bact.oalkyl.r<-(adonis.bc.bact.oalkyl.z)[[1]][["R2"]][[1]]
bact.oalkyl.r.df<-as.data.frame(bact.oalkyl.r)
# Extract p-values
bact.oalkyl.p<-(adonis.bc.bact.oalkyl.z)[[1]][["Pr(>F)"]][[1]]
bact.oalkyl.p.df<-as.data.frame(bact.oalkyl.p)

###Bacteria-Carboxyl
# Extract r2 value
bact.carboxyl.r<-(adonis.bc.bact.carboxyl.z)[[1]][["R2"]][[1]]
bact.carboxyl.r.df<-as.data.frame(bact.carboxyl.r)
# Extract p-values
bact.carboxyl.p<-(adonis.bc.bact.carboxyl.z)[[1]][["Pr(>F)"]][[1]]
bact.carboxyl.p.df<-as.data.frame(bact.carboxyl.p)

###Bacteria-C
# Extract r2 value
bact.c.r<-(adonis.bc.bact.c.z)[[1]][["R2"]][[1]]
bact.c.r.df<-as.data.frame(bact.c.r)
# Extract p-values
bact.c.p<-(adonis.bc.bact.c.z)[[1]][["Pr(>F)"]][[1]]
bact.c.p.df<-as.data.frame(bact.c.p)

###Bacteria-N
# Extract r2 value
bact.n.r<-(adonis.bc.bact.n.z)[[1]][["R2"]][[1]]
bact.n.r.df<-as.data.frame(bact.n.r)
# Extract p-values
bact.n.p<-(adonis.bc.bact.n.z)[[1]][["Pr(>F)"]][[1]]
bact.n.p.df<-as.data.frame(bact.n.p)

### Table

###------R2

#Merge all R2-values into a single column dataframe
bact.adonis.env.r<-Reduce(function(x, y) merge(x, y, all=TRUE), list(bact.moisture.r.df,bact.ph.r.df,bact.alkyl.r.df,bact.oalkyl.r.df,bact.carboxyl.r.df,bact.c.r.df,bact.n.r.df))
bact.adonis.env.r<-as.data.frame(t(bact.adonis.env.r))

#Create ID column from row names
bact.adonis.env.r$ID<-row.names(bact.adonis.env.r)

#Create new column without ".r" at end
ID.bact.r<-list(bact.moisture = 'bact.moisture.r',
                 bact.ph = 'bact.ph.r',
                 bact.alkyl = 'bact.alkyl.r',
                 bact.oalkyl = 'bact.oalkyl.r',
                 bact.carboxyl = 'bact.carboxyl.r',
                 bact.c = 'bact.c.r',
                 bact.n = 'bact.n.r')
IDs.bact.r<-names( ID.bact.r )
text.bact.r<-sapply(ID.bact.r, length)
IDMap.bact.r<-unlist( Map( rep, IDs.bact.r, text.bact.r ) )
names(IDMap.bact.r) <- unlist( ID.bact.r)
bact.adonis.env.r<-transform( bact.adonis.env.r, ID.bact.r = IDMap.bact.r[ID])

#Make ID.bact row names
row.names(bact.adonis.env.r)<-bact.adonis.env.r$ID.bact.r

#Rename column 1 (V1) to "P"
colnames(bact.adonis.env.r)[1] <- "R2"

#Remove ID column
bact.adonis.env.r<-bact.adonis.env.r[,c(1,3)]

#Rename column 2 to "ID"
colnames(bact.adonis.env.r)[2] <- "ID"

#Rename column 2 to "ID"
colnames(bact.adonis.env.r)[2] <- "ID"

###-------P-Values

#Merge all P-values into a single column dataframe
bact.adonis.env.p<-Reduce(function(x, y) merge(x, y, all=TRUE), list(bact.moisture.p.df,bact.ph.p.df,bact.alkyl.p.df,bact.oalkyl.p.df,bact.carboxyl.p.df,bact.c.p.df,bact.n.p.df))

#Transform data frame
bact.adonis.env.p<-as.data.frame(t(bact.adonis.env.p))

#Create ID column from row names
bact.adonis.env.p$ID<-row.names(bact.adonis.env.p)

#Create new column without ".p" at end
ID.bact.p<-list(bact.moisture = 'bact.moisture.p',
                 bact.ph = 'bact.ph.p',
                 bact.alkyl = 'bact.alkyl.p',
                 bact.oalkyl = 'bact.oalkyl.p',
                 bact.carboxyl = 'bact.carboxyl.p',
                 bact.c = 'bact.c.p',
                 bact.n = 'bact.n.p')
IDs.bact.p<-names( ID.bact.p )
text.bact.p<-sapply(ID.bact.p, length)
IDMap.bact.p<-unlist( Map( rep, IDs.bact.p, text.bact.p ) )
names(IDMap.bact.p) <- unlist( ID.bact.p)
bact.adonis.env.p<-transform( bact.adonis.env.p, ID.bact.p = IDMap.bact.p[ID])

#Make ID.bact row names
row.names(bact.adonis.env.p)<-bact.adonis.env.p$ID.bact.p

#Rename column 1 (V1) to "P"
colnames(bact.adonis.env.p)[1] <- "P"

#Remove ID column
bact.adonis.env.p<-bact.adonis.env.p[,c(1,3)]

#Rename column 2 to "ID"
colnames(bact.adonis.env.p)[2] <- "ID"

###-------Merging R2 and P together

###Merge R2 and P columns into single data frame
bact.adonis.env.final<-merge(bact.adonis.env.r,bact.adonis.env.p, by="ID",all=TRUE)

#Make ID row names
row.names(bact.adonis.env.final)<-bact.adonis.env.final$ID

#Remove ID column
bact.adonis.env.final<-bact.adonis.env.final[2:3]

#Round values
bact.adonis.env.final$R2<-(bact.adonis.env.final$R2*100)
bact.adonis.env.final$R2<-round(bact.adonis.env.final$R2,1)

bact.adonis.env.final$P<-round(bact.adonis.env.final$P,3)

# Sort in decending order from R2 values
bact.adonis.env.final<-bact.adonis.env.final[order(bact.adonis.env.final$R2,decreasing = TRUE),]

# Put P-values in parantheses
bact.adonis.env.final$P<-paste("(",bact.adonis.env.final$P,")",sep="")

# Combine R2 with P-values in single column
bact.adonis.env.final$`R2 (P)`<-paste(bact.adonis.env.final$R2,bact.adonis.env.final$P,sep=" ")

#Keep only the combined column
bact.adonis.env.final<-bact.adonis.env.final[3]

# Rename rows in the data frame to environmental variables
row.names(bact.adonis.env.final)<-c("Moisture","pH","Carboxyl","N","C","Alkyl","O-alkyl")

# Set pander options (table formatting)
set.alignment('right',row.names = 'left')
pander(bact.adonis.env.final,caption = "Table 3. PERMANOVA between BACTERIAL beta-diversity (Bray-Curtis dissimilarity) and measured environmental factors (z-score transformed).")
```

####FUNGI
```{r Beta_Fungi_Env}
# PERMANOVA - Fungi with Env Factors

#Fungi - Moisture
adonis.bc.fungi.moisture.z<-adonis(fungi.bc.dist~Moisture, data=new.env.qual.z)

#Fungi - pH
adonis.bc.fungi.ph.z<-adonis(fungi.bc.dist~pH, data=new.env.qual.z)

#Fungi - Alkyl
adonis.bc.fungi.alkyl.z<-adonis(fungi.bc.dist~Alkyl, data=new.env.qual.z)

#Fungi - O-alkyl
adonis.bc.fungi.oalkyl.z<-adonis(fungi.bc.dist~O_alkyl, data=new.env.qual.z)

#Fungi - Carboxyl
adonis.bc.fungi.carboxyl.z<-adonis(fungi.bc.dist~Carboxyl, data=new.env.qual.z)

#Fungi - C
adonis.bc.fungi.c.z<-adonis(fungi.bc.dist~C, data=new.env.qual.z)

#Fungi - N
adonis.bc.fungi.n.z<-adonis(fungi.bc.dist~N, data=new.env.qual.z)

###------------Extracting R2 and P-Values

###Fungi-Moisture
# Extract r2 value
fungi.moisture.r<-(adonis.bc.fungi.moisture.z)[[1]][["R2"]][[1]]
fungi.moisture.r.df<-as.data.frame(fungi.moisture.r)
# Extract p-values
fungi.moisture.p<-(adonis.bc.fungi.moisture.z)[[1]][["Pr(>F)"]][[1]]
fungi.moisture.p.df<-as.data.frame(fungi.moisture.p)

###Fungi-pH
# Extract r2 value
fungi.ph.r<-(adonis.bc.fungi.ph.z)[[1]][["R2"]][[1]]
fungi.ph.r.df<-as.data.frame(fungi.ph.r)
# Extract p-values
fungi.ph.p<-(adonis.bc.fungi.ph.z)[[1]][["Pr(>F)"]][[1]]
fungi.ph.p.df<-as.data.frame(fungi.ph.p)

###Fungi-Alkyl
# Extract r2 value
fungi.alkyl.r<-(adonis.bc.fungi.alkyl.z)[[1]][["R2"]][[1]]
fungi.alkyl.r.df<-as.data.frame(fungi.alkyl.r)
# Extract p-values
fungi.alkyl.p<-(adonis.bc.fungi.alkyl.z)[[1]][["Pr(>F)"]][[1]]
fungi.alkyl.p.df<-as.data.frame(fungi.alkyl.p)

###Fungi-O_alkyl
# Extract r2 value
fungi.oalkyl.r<-(adonis.bc.fungi.oalkyl.z)[[1]][["R2"]][[1]]
fungi.oalkyl.r.df<-as.data.frame(fungi.oalkyl.r)
# Extract p-values
fungi.oalkyl.p<-(adonis.bc.fungi.oalkyl.z)[[1]][["Pr(>F)"]][[1]]
fungi.oalkyl.p.df<-as.data.frame(fungi.oalkyl.p)

###Fungi-Carboxyl
# Extract r2 value
fungi.carboxyl.r<-(adonis.bc.fungi.carboxyl.z)[[1]][["R2"]][[1]]
fungi.carboxyl.r.df<-as.data.frame(fungi.carboxyl.r)
# Extract p-values
fungi.carboxyl.p<-(adonis.bc.fungi.carboxyl.z)[[1]][["Pr(>F)"]][[1]]
fungi.carboxyl.p.df<-as.data.frame(fungi.carboxyl.p)

###Fungi-C
# Extract r2 value
fungi.c.r<-(adonis.bc.fungi.c.z)[[1]][["R2"]][[1]]
fungi.c.r.df<-as.data.frame(fungi.c.r)
# Extract p-values
fungi.c.p<-(adonis.bc.fungi.c.z)[[1]][["Pr(>F)"]][[1]]
fungi.c.p.df<-as.data.frame(fungi.c.p)

###Fungi-N
# Extract r2 value
fungi.n.r<-(adonis.bc.fungi.n.z)[[1]][["R2"]][[1]]
fungi.n.r.df<-as.data.frame(fungi.n.r)
# Extract p-values
fungi.n.p<-(adonis.bc.fungi.n.z)[[1]][["Pr(>F)"]][[1]]
fungi.n.p.df<-as.data.frame(fungi.n.p)

### Table

###------R2

#Merge all R2-values into a single column dataframe
fungi.adonis.env.r<-Reduce(function(x, y) merge(x, y, all=TRUE), list(fungi.moisture.r.df,fungi.ph.r.df,fungi.alkyl.r.df,fungi.oalkyl.r.df,fungi.carboxyl.r.df,fungi.c.r.df,fungi.n.r.df))
fungi.adonis.env.r<-as.data.frame(t(fungi.adonis.env.r))

#Create ID column from row names
fungi.adonis.env.r$ID<-row.names(fungi.adonis.env.r)

#Create new column without ".r" at end
ID.fungi.r<-list(fungi.moisture = 'fungi.moisture.r',
                 fungi.ph = 'fungi.ph.r',
                 fungi.alkyl = 'fungi.alkyl.r',
                 fungi.oalkyl = 'fungi.oalkyl.r',
                 fungi.carboxyl = 'fungi.carboxyl.r',
                 fungi.c = 'fungi.c.r',
                 fungi.n = 'fungi.n.r')
IDs.fungi.r<-names( ID.fungi.r )
text.fungi.r<-sapply(ID.fungi.r, length)
IDMap.fungi.r<-unlist( Map( rep, IDs.fungi.r, text.fungi.r ) )
names(IDMap.fungi.r) <- unlist( ID.fungi.r)
fungi.adonis.env.r<-transform( fungi.adonis.env.r, ID.fungi.r = IDMap.fungi.r[ID])

#Make ID.fungi row names
row.names(fungi.adonis.env.r)<-fungi.adonis.env.r$ID.fungi.r

#Rename column 1 (V1) to "P"
colnames(fungi.adonis.env.r)[1] <- "R2"

#Remove ID column
fungi.adonis.env.r<-fungi.adonis.env.r[,c(1,3)]

#Rename column 2 to "ID"
colnames(fungi.adonis.env.r)[2] <- "ID"

#Rename column 2 to "ID"
colnames(fungi.adonis.env.r)[2] <- "ID"

###-------P-values

#Merge all P-values into a single column dataframe
fungi.adonis.env.p<-Reduce(function(x, y) merge(x, y, all=TRUE), list(fungi.moisture.p.df,fungi.ph.p.df,fungi.alkyl.p.df,fungi.oalkyl.p.df,fungi.carboxyl.p.df,fungi.c.p.df,fungi.n.p.df))

#Transform data frame
fungi.adonis.env.p<-as.data.frame(t(fungi.adonis.env.p))

#Create ID column from row names
fungi.adonis.env.p$ID<-row.names(fungi.adonis.env.p)

#Create new column without ".p" at end
ID.fungi.p<-list(fungi.moisture = 'fungi.moisture.p',
                 fungi.ph = 'fungi.ph.p',
                 fungi.alkyl = 'fungi.alkyl.p',
                 fungi.oalkyl = 'fungi.oalkyl.p',
                 fungi.carboxyl = 'fungi.carboxyl.p',
                 fungi.c = 'fungi.c.p',
                 fungi.n = 'fungi.n.p')
IDs.fungi.p<-names( ID.fungi.p )
text.fungi.p<-sapply(ID.fungi.p, length)
IDMap.fungi.p<-unlist( Map( rep, IDs.fungi.p, text.fungi.p ) )
names(IDMap.fungi.p) <- unlist( ID.fungi.p)
fungi.adonis.env.p<-transform( fungi.adonis.env.p, ID.fungi.p = IDMap.fungi.p[ID])

#Make ID.fungi row names
row.names(fungi.adonis.env.p)<-fungi.adonis.env.p$ID.fungi.p

#Rename column 1 (V1) to "P"
colnames(fungi.adonis.env.p)[1] <- "P"

#Remove ID column
fungi.adonis.env.p<-fungi.adonis.env.p[,c(1,3)]

#Rename column 2 to "ID"
colnames(fungi.adonis.env.p)[2] <- "ID"

###-------Merging R2 and P together

###Merge R2 and P columns into single data frame
fungi.adonis.env.final<-merge(fungi.adonis.env.r,fungi.adonis.env.p, by="ID",all=TRUE)

#Make ID row names
row.names(fungi.adonis.env.final)<-fungi.adonis.env.final$ID

#Remove ID column
fungi.adonis.env.final<-fungi.adonis.env.final[2:3]

#Round values
fungi.adonis.env.final$R2<-(fungi.adonis.env.final$R2*100)
fungi.adonis.env.final$R2<-round(fungi.adonis.env.final$R2,1)

fungi.adonis.env.final$P<-round(fungi.adonis.env.final$P,3)

# Sort in decending order from R2 values
fungi.adonis.env.final<-fungi.adonis.env.final[order(fungi.adonis.env.final$R2,decreasing = TRUE),]

# Put P-values in parantheses
fungi.adonis.env.final$P<-paste("(",fungi.adonis.env.final$P,")",sep="")

# Combine R2 with P-values in single column
fungi.adonis.env.final$`R2 (P)`<-paste(fungi.adonis.env.final$R2,fungi.adonis.env.final$P,sep=" ")

#Keep only the combined column
fungi.adonis.env.final<-fungi.adonis.env.final[3]

# Rename rows in the data frame to environmental variables
row.names(fungi.adonis.env.final)<-c("Moisture","pH","N","Carboxyl","C","Alkyl","O-alkyl")

# Set pander options (table formatting)
set.alignment('right',row.names = 'left')
pander(fungi.adonis.env.final,caption = "Table 3. PERMANOVA between FUNGAL beta-diversity (Bray-Curtis dissimilarity) and measured environmental factors (z-score transformed).")
```

###Visualizing Beta Diversity Through Unconstrained Ordination - NMDS
```{r NMDS_Bacteria}
# NMDS Ordination for Bacteria
ord1<-metaMDS(bacteria, trymax=1000, k=3)
ord1
scores(ord1)
stressplot(ord1)

# Data for plotting
bact.NMDS.data<-new.env
bact.NMDS.data$NMDS1<-ord1$points[ ,1]
bact.NMDS.data$NMDS2<-ord1$points[ ,2]

# R2 Superscript for plot
bact.ss<-0.859
bact.ss<-paste("R^2 == ", round(bact.ss,2))
```

```{r NMDS_Fungi}
# NMDS Ordination for Fungi
ord2<-metaMDS(fungi, trymax=1000, k=3)
ord2
scores(ord2)
stressplot(ord2)

# Data for plotting
fungi.NMDS.data<-new.env
fungi.NMDS.data$NMDS1<-ord2$points[ ,1]
fungi.NMDS.data$NMDS2<-ord2$points[ ,2]

# R2 Superscript
fung.ss<-0.904
fung.ss<-paste("R^2 == ", round(fung.ss,2))
```

###Reproducing Figure 1 - NMDS
```{r Fig_1}
# Set geom_point shape to new default prior to plotting
update_geom_defaults("point", list(shape=21))

# Plotting bacteria NMDS in ggplot2
bact.ggplot<-ggplot(bact.NMDS.data, aes(x=NMDS1, y=NMDS2)) + aes(shape=factor(Site)) + geom_point(colour="black", size=3.5, aes(fill=factor(Community))) + scale_fill_manual(values=c("white", "black")) + scale_shape_manual("Site", values=c(21,22,23,24))+ scale_colour_manual(values=c("black", "black")) + labs(fill="Community") + theme_bw() + scale_x_continuous(limits=c(-0.5, 0.5)) + scale_y_continuous(limits=c(-0.55, 0.45)) + annotate("text", x=-0.48, y=0.45, label = "(a)", size=6) + theme(legend.key=element_blank()) + annotate("text", x=-0.5, y=-0.50, label=bact.ss, parse=TRUE, hjust=0, size=4) + annotate("text", x=-0.5, y=-0.55, label="Stress=0.14", hjust=0, size=4) + theme(axis.ticks.y = element_blank(), axis.ticks.x = element_blank(), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank())

# Plotting fungi NMDS in ggplot2
fungi.ggplot<-ggplot(fungi.NMDS.data, aes(x=NMDS1, y=NMDS2)) + aes(shape=factor(Site)) + geom_point(colour="black", size=3.5, aes(fill=factor(Community))) + scale_fill_manual(values=c("white", "black")) + scale_shape_manual("Site", values=c(21,22,23,24)) + scale_colour_manual(values=c("black", "black")) + labs(fill="Community") + theme_bw() + scale_x_continuous(limits=c(-0.6, 0.6)) + scale_y_continuous(limits=c(-0.55, 0.45)) + annotate("text", x=-0.58, y=0.45, label = "(b)", size=6) + theme(legend.key=element_blank()) + annotate("text", x=-0.6, y=-0.5, label=fung.ss, parse=TRUE, hjust=0, size=4) + annotate("text", x=-0.6, y=-0.55, label="Stress=0.13", hjust=0, size=4) + theme(axis.ticks.y = element_blank(), axis.ticks.x = element_blank(), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank())
```

```{r echo=FALSE}
#Plotting multiple plots on the same page

#Add common legend for multiple ggplot2 graphs

get_legend<-function(myggplot){ 
	tmp<-ggplot_gtable(ggplot_build(myggplot))
	leg<-which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
	legend<-tmp$grobs[[leg]]
	return(legend)
	}
	
#Step 1 - Create plots
bact.ggplot2<-ggplot(bact.NMDS.data, aes(x=NMDS1, y=NMDS2)) + aes(shape=factor(Site)) + geom_point(colour="black", size=3.5, aes(fill=factor(Community))) + scale_fill_manual(values=c("white", "black")) + scale_shape_manual("Site", values=c(21,22,23,24))+ scale_colour_manual(values=c("black", "black")) + labs(fill="Community") + theme_bw() + scale_x_continuous(limits=c(-0.5, 0.5)) + scale_y_continuous(limits=c(-0.55, 0.45)) + annotate("text", x=-0.48, y=0.45, label = "(a)", size=6) + theme(legend.key=element_blank()) + annotate("text", x=-0.5, y=-0.50, label=bact.ss, parse=TRUE, hjust=0, size=4) + annotate("text", x=-0.5, y=-0.55, label="Stress=0.14", hjust=0, size=4) + theme(axis.ticks.y = element_blank(), axis.ticks.x = element_blank(), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + xlab("NMDS1 (14.9% variation)") + ylab("NMDS2 (4.7% variation)")

fungi.ggplot2<-ggplot(fungi.NMDS.data, aes(x=NMDS1, y=NMDS2)) + aes(shape=factor(Site)) + geom_point(colour="black", size=3.5, aes(fill=factor(Community))) + scale_fill_manual(values=c("white", "black")) + scale_shape_manual("Site", values=c(21,22,23,24)) + scale_colour_manual(values=c("black", "black")) + labs(fill="Community") + theme_bw() + scale_x_continuous(limits=c(-0.6, 0.6)) + scale_y_continuous(limits=c(-0.55, 0.45)) + annotate("text", x=-0.58, y=0.45, label = "(b)", size=6) + theme(legend.position="none") + theme(legend.key=element_blank()) + annotate("text", x=-0.6, y=-0.5, label=fung.ss, parse=TRUE, hjust=0, size=4) + annotate("text", x=-0.6, y=-0.55, label="Stress=0.13", hjust=0, size=4) + theme(axis.ticks.y = element_blank(), axis.ticks.x = element_blank(), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + xlab("NMDS1 (22.8% variation)") + ylab("NMDS2 (5.5% variation)")

#2. Save the legend
legend<-get_legend(bact.ggplot2)

#3. Remove legend from bact
bact.ggplot2<-bact.ggplot2 + theme(legend.position="none")
```

```{r fig.width=12}
# Arrange ggplot2 graphs with a specific width
grid.arrange(bact.ggplot2, fungi.ggplot2, legend, ncol=3, widths=c(5,5,2))
```
**Figure 1.** Nonmetric multidimensional scaling (NMDS) ordinations based on Bray-Curtis dissimilarity of OTU abundance between (a) bacterial and (b) fungal communities reveals spatial variation by site and between total and active communities. Each marker represents an individual plot in the NMDS charts.

###Taxonomy - Bacterial Community Relative Abundance - Order-Level
```{r Taxa_Bacteria}
# Call in bacteria taxonomy table (absolute abundance)
bact.tax<-read.table("c.g.DNA.16S.combined.trim.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.03.subsample.0.03.cons.tax.summary", header = TRUE, sep = "\t")

# Subset taxonomy file to pull out only order level taxa ("taxlevel" = 4)
bact.order<-subset(bact.tax, taxlevel==4)

# Remove N-deposition plots
bact.order<-bact.order[, -c(4:5,9:11,15:17,21:23,27:29,33:35,39:41,45:47,51:54)]

# Create new column with rankID and taxon
bact.order$taxID<-paste(bact.order$taxon,bact.order$rankID,sep="_")

# Preserve taxID as row names
row.names(bact.order)<-bact.order$taxID

# Remove non-numeric columns
bact.order.new<-bact.order[,-c(1:3,28)]

# Convert from absolute to relative abundance
bact.order.sums<-colSums(bact.order.new)
bact.order.test1<-as.matrix(bact.order.new)
bact.order.test2<-1/bact.order.sums
bact.order.test3<-t(bact.order.test1)
bact.order.test4<-(bact.order.test3*bact.order.test2)*100
bact.order.test5<-t(bact.order.test4)
bact.order.rel<-as.data.frame(bact.order.test5)
bact.order.rel$taxID<-row.names(bact.order.rel)

# Add factor columns back in
bact.order.factor<-bact.order[,c(1:3,28)]

# Merge factors back into Rel Abund table
bact.order.final<-merge(bact.order.factor,bact.order.rel,by="taxID",all=TRUE)

# Remove certain factors for downstream processing
bact.order.final<-bact.order.final[,-c(2:3)]
```

#### Separate Active Bacteria
```{r Taxa_Bacteria_Active}
# Active data frame
bact.order.active<-bact.order.final[,1:14]

# Calculate taxon means and add new means column
bact.order.active$MEANS_Active<-rowMeans(bact.order.active[,3:14])

# Calculate taxon sd and add new sd column
bact.order.active$SD_Active<-apply(as.matrix(bact.order.active[,3:14]),1,sd)

# Preserve taxID as row names
row.names(bact.order.active)<-bact.order.active$taxID

# Remove non-numeric columns
bact.order.active.new<-bact.order.active[,-1]

# Collect all taxon greater than 1% rel abund
bact.order.active.relevant<-subset(bact.order.active.new, MEANS_Active>=1 | taxon=="Flavobacteriales")

bact.order.active.relevant<-subset(bact.order.active.relevant, taxon!="unclassified")

#Remove taxon column
bact.order.active.relevant<-bact.order.active.relevant[,-1]

# Collect all taxon less than 1% rel abund
bact.order.active.other<-subset(bact.order.active.new, MEANS_Active<1 | taxon=="unclassified")

bact.order.active.other<-subset(bact.order.active.other, taxon!="Flavobacteriales")

# Sum "unclassfied" into single row
bact.order.active.other<-ddply(bact.order.active.other, "taxon", numcolwise(sum))

# Pull "unclassified" from "other" table
bact.order.active.other.noclass<-subset(bact.order.active.other, taxon!="unclassified")

# Create "unclassified" factor to pull out these taxa from "other" table
bact.active.unclass<-"unclassified"
bact.order.active.other.unclass<-bact.order.active.other[bact.order.active.other$taxon %in% bact.active.unclass,]

# Preserve taxon as row names
row.names(bact.order.active.other.noclass)<-bact.order.active.other.noclass$taxon
row.names(bact.order.active.other.unclass)<-bact.order.active.other.unclass$taxon

# Save names of all taxa placed in "Other" category
bact.other.names<-row.names(bact.order.active.other.noclass)

#Remove taxon column
bact.order.active.other.noclass<-bact.order.active.other.noclass[,-1]
bact.order.active.other.unclass<-bact.order.active.other.unclass[,-1]

# Column sums
bact.order.active.other.vec<-colSums(bact.order.active.other.noclass)

# Bind "Other" to relevant data frame
bact.order.active.final<-rbind(bact.order.active.relevant,bact.order.active.other.vec,bact.order.active.other.unclass)

# Rename row names with "Other" column
row.names(bact.order.active.final)<-c("Acidobacteria_Gp1","Actinomycetales","Burkholderiales","Caulobacterales","Flavobacteriales","Planctomycetales","Pseudomonadales","Rhizobiales","Rhodospirillales","Sphingobacteriales","Sphingomonadales","Other","Unclassified")

# Add back in Taxon column
bact.order.active.final$Taxon<-row.names(bact.order.active.final)
```

#### Separate Total Bacteria
```{r Taxa_Bacteria_Total}
# Total data frame
bact.order.total<-bact.order.final[,c(1:2,15:26)]

# Calculate taxon means and add new means column
bact.order.total$MEANS_Total<-rowMeans(bact.order.total[,3:14])

# Calculate taxon sd and add new sd column
bact.order.total$SD_Total<-apply(as.matrix(bact.order.total[,3:14]),1,sd)

# Preserve taxID as row names
row.names(bact.order.total)<-bact.order.total$taxID

# Remove non-numeric columns
bact.order.total.new<-bact.order.total[,-1]

# Collect all taxon greater than 1% rel abund
bact.order.total.relevant<-subset(bact.order.total.new, MEANS_Total>=1)

# Remove unclassified taxa
bact.order.total.relevant<-subset(bact.order.total.relevant, taxon!="unclassified")

# Remove TM7 taxa
bact.order.total.relevant<-subset(bact.order.total.relevant, taxon!="TM7_order_incertae_sedis")

# Remove taxon column
bact.order.total.relevant<-bact.order.total.relevant[,-1]

# Collect all taxon less than 1% rel abund including all unclassified and TM7
bact.order.total.other<-subset(bact.order.total.new, MEANS_Total<1 | taxon=="unclassified" | taxon=="TM7_order_incertae_sedis")

# Sum "unclassfied" into single row
bact.order.total.other<-ddply(bact.order.total.other, "taxon", numcolwise(sum))

# Pull "unclassified" from "other" table
bact.order.total.other.noclass<-subset(bact.order.total.other, taxon!="unclassified")

# Create "unclassified" object to pull these taxa out of "other" table
bact.total.unclass<-"unclassified"
bact.order.total.other.unclass<-bact.order.total.other[bact.order.total.other$taxon %in% bact.total.unclass,]

# Preserve taxon as row names
row.names(bact.order.total.other.noclass)<-bact.order.total.other.noclass$taxon
row.names(bact.order.total.other.unclass)<-bact.order.total.other.unclass$taxon

#Remove taxon column
bact.order.total.other.noclass<-bact.order.total.other.noclass[,-1]
bact.order.total.other.unclass<-bact.order.total.other.unclass[,-1]

# Column sums
bact.order.total.other.vec<-colSums(bact.order.total.other.noclass)

# Bind "Other" to relevant data frame
bact.order.total.final<-rbind(bact.order.total.relevant,bact.order.total.other.vec,bact.order.total.other.unclass)

# Rename row names with "Other" column
row.names(bact.order.total.final)<-c("Acidobacteria_Gp1","Actinomycetales","Burkholderiales","Caulobacterales","Flavobacteriales","Planctomycetales","Pseudomonadales","Rhizobiales","Rhodospirillales","Sphingobacteriales","Sphingomonadales","Other","Unclassified")

# Add back in Taxon column
bact.order.total.final$Taxon<-row.names(bact.order.total.final)

# Merge Active and Total data frames
bact.order.merge<-merge(bact.order.total.final,bact.order.active.final,by="Taxon",all=TRUE)
```

###MANOVA on change in relative abundance between bacterial communities at the order-level
```{r Taxa_MANOVA_Bacteria}
# Final Bacteria Taxonomy table for MANOVA
bact.tax.final<-bact.order.merge[, -c(14,15,28,29)]

### Transpose taxonomy table

# Remember taxon as non-numeric values
bact.tax.final.taxon<-bact.tax.final$Taxon

# Transpose all but first column (Taxon)
bact.tax.final<-as.data.frame(t(bact.tax.final[,-1]))
colnames(bact.tax.final)<-bact.tax.final.taxon

# Add new columns for "Site" and "Community" to the end of the data table
bact.tax.final$Site <- c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")

bact.tax.final$Community <- c("Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active")

#Rename "Acidobacteria Gp1" without spaces
names(bact.tax.final)[names(bact.tax.final)=='Acidobacteria Gp1']<-'Acidobacteria_Gp1'

#Create matrix of numeric columns
bact.bind.mat<-as.matrix(cbind(bact.tax.final[,1:13]))

# MANOVA
bact.manova<-manova(bact.bind.mat~Community, data=bact.tax.final)
bact.man.sum<-summary.aov(bact.manova)
bact.man.sum

# Extract p-values
bact.man.p<-lapply(bact.man.sum, function(i) {
  i[1,5]
})

# Make into df
bact.man.p.df<-as.data.frame(do.call(rbind,bact.man.p))

# Clean up the format of df
colnames(bact.man.p.df)<-"P-Value"

# Get rid of "Response"
bact.man.p.df$ID<-row.names(bact.man.p.df)

bact.man.p.df$Taxon<-gsub(" Response ","",bact.man.p.df$ID)

# Make Taxon row names
row.names(bact.man.p.df)<-bact.man.p.df$Taxon

# Remove ID and Taxon columns
bact.man.p.df<-subset(bact.man.p.df,select="P-Value")
```

###SIMPER Analysis for Taxon Contributions to Dissimilarity
```{r Taxa_Bacteria_SIMPER}
#Preserve Taxon as row names
bact.order.simper<-bact.order.merge

# Remove non-numeric column and Means and SD columns
bact.order.simper<-bact.order.simper[,-c(14,15,28,29)]

### Transpose taxonomy table

# Remember taxon as non-numeric values
bact.order.simper.taxon<-bact.order.simper$Taxon

# Transpose all but first column (Taxon)
bact.order.simper<-as.data.frame(t(bact.order.simper[,-1]))
colnames(bact.order.simper)<-bact.order.simper.taxon

# Run SIMPER Analysis
bact.sim<-with(new.env, simper(bact.order.simper, Community))
summary(bact.sim)

# Convert to Dataframe
bact.sim.df<-data.frame(bact.sim$Total_Active$species,bact.sim$Total_Active$average,bact.sim$Total_Active$ava,bact.sim$Total_Active$avb, stringsAsFactors = FALSE)

# Calculate overall % Dissimilarity between Total and Active communities
sum(bact.sim.df$bact.sim.Total_Active.average)*100

# Set Overall contribution values aside
bact.sim.overall<-bact.sim$Total_Active$overall

# Calculate taxon contributions to dissimilarity
bact.sim.df$Contribution<-(bact.sim.df$bact.sim.Total_Active.average/bact.sim.overall)*100

# Pull out relevant columns
bact.sim.final<-subset(bact.sim.df,select = c("bact.sim.Total_Active.species","Contribution"))

# Rename column 1 to Taxon
colnames(bact.sim.final)<-c("Taxon","SIMPER")
```

###Recreating Table 4. Relative abundance of total and active bacteria with % difference between communities and SIMPER % dissimillarity of taxa between communities.
```{r Table_4_Bacteria}
# Final Bacteria Means table (Table 3 recreated)
bact.tax.final.means<-subset(bact.order.merge, select=c("Taxon","MEANS_Total","SD_Total","MEANS_Active","SD_Active"))

# Add in SIMPER Contribution column to final table
bact.tax.final.means<-merge(bact.tax.final.means,bact.sim.final,by="Taxon",all=TRUE)

# Sort in decending order from Total community
bact.tax.final.means<-bact.tax.final.means[order(bact.tax.final.means$MEANS_Total,decreasing = TRUE),]

# Round numbers in each column
bact.tax.final.means$MEANS_Total<-round(bact.tax.final.means$MEANS_Total,0)

bact.tax.final.means$MEANS_Active<-round(bact.tax.final.means$MEANS_Active,0)

bact.tax.final.means$SD_Total<-round(bact.tax.final.means$SD_Total,1)

bact.tax.final.means$SD_Active<-round(bact.tax.final.means$SD_Active,1)

# Calculate difference between Total and Active Means
bact.tax.final.means$Difference<-bact.tax.final.means$MEANS_Active - bact.tax.final.means$MEANS_Total

bact.tax.final.means$Difference<-round(bact.tax.final.means$Difference,0)

# Preserve taxon as row names
row.names(bact.tax.final.means)<-bact.tax.final.means$Taxon
bact.tax.final.means<-bact.tax.final.means[,-1]

# Paste Total means with sd into single column
bact.tax.final.means$Total<-paste(bact.tax.final.means$MEANS_Total,"±",bact.tax.final.means$SD_Total,sep=" ")

# Paste Active means with sd into single column
bact.tax.final.means$Active<-paste(bact.tax.final.means$MEANS_Active,"±",bact.tax.final.means$SD_Active,sep=" ")

# Subset out columns of interest
bact.table2<-subset(bact.tax.final.means,select=c("Total","Active","Difference","SIMPER"))

# Round SIMPER values
bact.table2$SIMPER<-round(bact.table2$SIMPER,0)

# Merging everything into Table 3
bact.table2$Taxon<-row.names(bact.table2)
bact.man.p.df$Taxon<-row.names(bact.man.p.df)

#Final table
bact.table2.final<-merge(bact.table2,bact.man.p.df,by="Taxon",all=TRUE)

# Round p-values to 2 decimals
bact.table2.final$`P-Value`<-round(bact.table2.final$`P-Value`,3)

# Put P-values in parantheses
bact.table2.final$P2<-paste("(",bact.table2.final$`P-Value`,")",sep="")

# Add + signs to Difference factors
for(i in 1:length(bact.table2.final$Taxon)) {
  if (bact.table2.final[i,4]>0) {
    bact.table2.final[i,4]<-paste("+",bact.table2.final[i,4],sep="")}
}

# Combine Difference with P-values in single column
bact.table2.final$`Difference % (P-Value)`<-paste(bact.table2.final$Difference,bact.table2.final$P2,sep=" ")

# Remove unnecessary columns from final table
bact.table2.final<-bact.table2.final[,-c(4,6,7)]

bact.table2.final<-bact.table2.final[c("Taxon","Total","Active","Difference % (P-Value)","SIMPER")]

# Reorder Taxon by Total Rel Abund
bact.table2.final$Taxon<-as.factor(bact.table2.final$Taxon)
bact.table2.final$Taxon<-reorder(bact.table2.final$Taxon,c(3,1,5,6,10,13,7,11,2,8,4,9,12))
bact.table2.final<-bact.table2.final[order(bact.table2.final$Taxon),]

# Name Taxon row name
row.names(bact.table2.final)<-bact.table2.final$Taxon

# Remove Taxon column
bact.table2.final<-bact.table2.final[,-1]

# Rename rows in the data frame to environmental variables
row.names(bact.table2.final)<-c("Actinomycetales","Rhizobiales","Acidobacteria Gp1","Sphingobacteriales","Burkholderiales","Caulobacterales","Planctomycetales","Rhodospirillales","Sphingomonadales","Flavobacteriales","Pseudomonadales","Unclassified","Other")

# Rename columns
colnames(bact.table2.final)<-c("% Total","% Active", "% Difference (P-value)","% Dissimilarity Contribution")

# Set pander options (table formatting)
set.alignment('right',row.names = 'left')
pander(bact.table2.final,caption = "Table 4. Relative abundance of total and active bacterial taxa (mean ± standard deviation) with mean percent difference from total community.")
```

**Supplemental Table S3.** List of "Other" bacteria (<1% relative abundance)
```{r}
bact.other.names
```

###Taxonomy - Fungal Community Relative Abundance
```{r Taxa_Fungi}
# Call in fungal taxonomy table (absolute abundance)
fungi.tax<-read.table("c.g.DNA.28S.combined.trim.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.01.subsample.0.01.cons.tax.summary", header = TRUE, sep = "\t")

# Subset taxonomy file to pull out only class level taxa ("taxlevel" = 4)
fungi.phyla<-subset(fungi.tax, taxlevel==4)

# Remove N-deposition plot columns
fungi.phyla<-fungi.phyla[, -c(4:5,9:11,15:17,21:23,27:29,33:35,39:41,45:47,51:54)]

# Create new column with rankID and taxon
fungi.phyla$taxID<-paste(fungi.phyla$taxon,fungi.phyla$rankID,sep="_")

# Preserve taxID as row names
row.names(fungi.phyla)<-fungi.phyla$taxID

# Remove non-numeric columns
fungi.phyla.new<-fungi.phyla[,-c(1:3,28)]

# Convert from absolute to relative abundance
fungi.sums<-colSums(fungi.phyla.new)
fungi.test1<-as.matrix(fungi.phyla.new)
fungi.test2<-1/fungi.sums
fungi.test3<-t(fungi.test1)
fungi.test4<-(fungi.test3*fungi.test2)*100
fungi.test5<-t(fungi.test4)
fungi.phyla.rel<-as.data.frame(fungi.test5)
fungi.phyla.rel$taxID<-row.names(fungi.phyla.rel)

# Add factor columns back in
fungi.phyla.factor<-fungi.phyla[,c(1:3,28)]

# Merge factors back into Rel Abund table
fungi.phyla.final<-merge(fungi.phyla.factor,fungi.phyla.rel,by="taxID",all=TRUE)

# Remove certain factors for downstream processing
fungi.phyla.final<-fungi.phyla.final[,-c(2:3)]
```

#### Subset Active Fungi
```{r Taxa_Fungi_Active}
# Active data frame
fungi.active<-fungi.phyla.final[,1:14]

# Calculate taxon means and add new means column
fungi.active$MEANS_Active<-rowMeans(fungi.active[,3:14])

# Calculate taxon sd and add new sd column
fungi.active$SD_Active<-apply(as.matrix(fungi.active[,3:14]),1,sd)

# Preserve taxID as row names
row.names(fungi.active)<-fungi.active$taxID

# Remove non-numeric columns
fungi.active.new<-fungi.active[,-1]

# Collect all taxon greater than 1% rel abund
fungi.active.relevant<-subset(fungi.active.new, MEANS_Active>=1)

fungi.active.relevant<-subset(fungi.active.relevant, taxon!="unclassified")

#Remove taxon column
fungi.active.relevant<-fungi.active.relevant[,-1]

# Collect all taxon less than 1% rel abund
fungi.active.other<-subset(fungi.active.new, MEANS_Active<1 | taxon=="unclassified")

# Sum "unclassfied" into single row
fungi.active.other<-ddply(fungi.active.other, "taxon", numcolwise(sum))

# Pull "unclassified" from "other" table
fungi.active.other.noclass<-subset(fungi.active.other, taxon!="unclassified")

# Create "unclassified" factor to pull out these taxa from "other" table
fungi.active.unclass<-"unclassified"
fungi.active.other.unclass<-fungi.active.other[fungi.active.other$taxon %in% fungi.active.unclass,]

# Preserve taxon as row names
row.names(fungi.active.other.noclass)<-fungi.active.other.noclass$taxon
row.names(fungi.active.other.unclass)<-fungi.active.other.unclass$taxon

# Save names of all taxa placed in "Other" category
fungi.other.names<-row.names(fungi.active.other.noclass)

#Remove taxon column
fungi.active.other.noclass<-fungi.active.other.noclass[,-1]
fungi.active.other.unclass<-fungi.active.other.unclass[,-1]

# Column sums
fungi.active.other.vec<-colSums(fungi.active.other.noclass)

# Bind "Other" and "unclassified" to relevant data frame
fungi.active.final<-rbind(fungi.active.relevant,fungi.active.other.vec,fungi.active.other.unclass)

# Rename row names with "Other" column
row.names(fungi.active.final)<-c("Agaricomycetes","Dothideomycetes","Eurotiomycetes","Leotiomycetes","Microbotryomycetes","Sordariomycetes","Tremellomycetes","Other","Unclassified")

# Add back in Taxon column
fungi.active.final$Taxon<-row.names(fungi.active.final)
```

#### Separate Total Fungi
```{r Taxa_Fungi_Total}
# Total data frame
fungi.total<-fungi.phyla.final[,c(1:2,15:26)]

# Calculate taxon means and add new means column
fungi.total$MEANS_Total<-rowMeans(fungi.total[,3:14])

# Calculate taxon sd and add new sd column
fungi.total$SD_Total<-apply(as.matrix(fungi.total[,3:14]),1,sd)

# Preserve taxID as row names
row.names(fungi.total)<-fungi.total$taxID

# Remove non-numeric columns
fungi.total.new<-fungi.total[,-1]

# Collect all taxon greater than 1% rel abund
fungi.total.relevant<-subset(fungi.total.new, MEANS_Total>=1)

fungi.total.relevant<-subset(fungi.total.relevant, taxon!="unclassified")

#Remove taxon column
fungi.total.relevant<-fungi.total.relevant[,-1]

# Collect all taxon less than 1% rel abund
fungi.total.other<-subset(fungi.total.new, MEANS_Total<1 | taxon=="unclassified")

# Sum "unclassfied" into single row
fungi.total.other<-ddply(fungi.total.other, "taxon", numcolwise(sum))

# Pull "unclassified" from "other" table
fungi.total.other.noclass<-subset(fungi.total.other, taxon!="unclassified")

# Create "unclassified" factor to pull out these taxa from "other" table
fungi.total.unclass<-"unclassified"
fungi.total.other.unclass<-fungi.total.other[fungi.total.other$taxon %in% fungi.total.unclass,]

# Preserve taxon as row names
row.names(fungi.total.other.noclass)<-fungi.total.other.noclass$taxon
row.names(fungi.total.other.unclass)<-fungi.total.other.unclass$taxon

#Remove taxon column
fungi.total.other.noclass<-fungi.total.other.noclass[,-1]
fungi.total.other.unclass<-fungi.total.other.unclass[,-1]

# Column sums
fungi.total.other.vec<-colSums(fungi.total.other.noclass)

# Bind "Other" and "unclassified" to relevant data frame
fungi.total.final<-rbind(fungi.total.relevant,fungi.total.other.vec,fungi.total.other.unclass)

# Rename row names with "Other" column
row.names(fungi.total.final)<-c("Agaricomycetes","Dothideomycetes","Eurotiomycetes","Leotiomycetes","Microbotryomycetes","Sordariomycetes","Tremellomycetes","Other","Unclassified")

# Add back in Taxon column
fungi.total.final$Taxon<-row.names(fungi.total.final)

# Merge Active and Total data frames
fungi.merge<-merge(fungi.total.final,fungi.active.final,by="Taxon",all=TRUE)
```

###MANOVA on change in relative abundance between fungal communities at the class-level
```{r Taxa_Fungi_MANOVA}
# Final Fungal taxonomy table for MANOVA
fungi.tax.final<-fungi.merge[, -c(14,15,28,29)]

### Transpose taxonomy table

# Remember taxon as non-numeric values
fungi.tax.final.taxon<-fungi.tax.final$Taxon

# Transpose all but first column (Taxon)
fungi.tax.final<-as.data.frame(t(fungi.tax.final[,-1]))
colnames(fungi.tax.final)<-fungi.tax.final.taxon

# Add new columns for "Site" and "Community" to the end of the data table
fungi.tax.final$Site <- c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")

fungi.tax.final$Community <- c("Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active")

#Create matrix of numeric columns
fungi.bind.mat<-as.matrix(cbind(fungi.tax.final[,1:9]))

# MANOVA
fungi.manova<-manova(fungi.bind.mat~Community, data=fungi.tax.final)
fungi.man.sum<-summary.aov(fungi.manova)
fungi.man.sum

# Extract p-values
fungi.man.p<-lapply(fungi.man.sum, function(i) {
  i[1,5]
})

# Make into df
fungi.man.p.df<-as.data.frame(do.call(rbind,fungi.man.p))

# Clean up the format of df
colnames(fungi.man.p.df)<-"P-Value"

# Get rid of "Response"
fungi.man.p.df$ID<-row.names(fungi.man.p.df)

fungi.man.p.df$Taxon<-gsub(" Response ","",fungi.man.p.df$ID)

# Make Taxon row names
row.names(fungi.man.p.df)<-fungi.man.p.df$Taxon

# Remove ID and Taxon columns
fungi.man.p.df<-subset(fungi.man.p.df,select="P-Value")
```

###SIMPER Analysis for Taxon Contributions to Dissimilarity
```{r Taxa_Fungi_SIMPER}
#Preserve Taxon as row names
fungi.simper<-fungi.merge

# Remove non-numeric column and Means and SD columns
fungi.simper<-fungi.simper[,-c(14,15,28,29)]

### Transpose taxonomy table

# Remember taxon as non-numeric values
fungi.simper.taxon<-fungi.simper$Taxon

# Transpose all but first column (Taxon)
fungi.simper<-as.data.frame(t(fungi.simper[,-1]))
colnames(fungi.simper)<-fungi.simper.taxon

# Run SIMPER Analysis
fungi.sim<-with(new.env, simper(fungi.simper, Community))
summary(fungi.sim)

# Convert to Dataframe
fungi.sim.df<-data.frame(fungi.sim$Total_Active$species,fungi.sim$Total_Active$average,fungi.sim$Total_Active$ava,fungi.sim$Total_Active$avb, stringsAsFactors = FALSE)

# Calculate overall % Dissimilarity between Total and Active communities
sum(fungi.sim.df$fungi.sim.Total_Active.average)*100

# Set Overall contribution values aside
fungi.sim.overall<-fungi.sim$Total_Active$overall

# Calculate taxon contributions to dissimilarity
fungi.sim.df$Contribution<-(fungi.sim.df$fungi.sim.Total_Active.average/fungi.sim.overall)*100

# Pull out relevant columns
fungi.sim.final<-subset(fungi.sim.df,select = c("fungi.sim.Total_Active.species","Contribution"))

# Rename column 1 to Taxon
colnames(fungi.sim.final)<-c("Taxon","SIMPER")
```

###Recreating Table 4. Relative abundance of total and active fungi with % difference between communities and SIMPER % dissimillarity of taxa between communities.
```{r Table_4_Fungi}
# Final Fungi Means table (Table 3 recreated)
fungi.tax.final.means<-subset(fungi.merge, select=c("Taxon","MEANS_Total","SD_Total","MEANS_Active","SD_Active"))

# Add in SIMPER Contribution column to final table
fungi.tax.final.means<-merge(fungi.tax.final.means,fungi.sim.final,by="Taxon",all=TRUE)

# Sort in decending order from Total community
fungi.tax.final.means<-fungi.tax.final.means[order(fungi.tax.final.means$MEANS_Total,decreasing = TRUE),]

# Round numbers in each column
fungi.tax.final.means$MEANS_Total<-round(fungi.tax.final.means$MEANS_Total,0)

fungi.tax.final.means$MEANS_Active<-round(fungi.tax.final.means$MEANS_Active,0)

fungi.tax.final.means$SD_Total<-round(fungi.tax.final.means$SD_Total,1)

fungi.tax.final.means$SD_Active<-round(fungi.tax.final.means$SD_Active,1)

# Calculate difference between Total and Active Means
fungi.tax.final.means$Difference<-fungi.tax.final.means$MEANS_Active - fungi.tax.final.means$MEANS_Total

fungi.tax.final.means$Difference<-round(fungi.tax.final.means$Difference,0)

# Preserve taxon as row names
row.names(fungi.tax.final.means)<-fungi.tax.final.means$Taxon
fungi.tax.final.means<-fungi.tax.final.means[,-1]

#Paste Total means with sd into single column
fungi.tax.final.means$Total<-paste(fungi.tax.final.means$MEANS_Total,"±",fungi.tax.final.means$SD_Total,sep=" ")

#Paste Active means with sd into single column
fungi.tax.final.means$Active<-paste(fungi.tax.final.means$MEANS_Active,"±",fungi.tax.final.means$SD_Active,sep=" ")

# Subset out columns of interest
fungi.table2<-subset(fungi.tax.final.means,select=c("Total","Active","Difference","SIMPER"))

# Round SIMPER values
fungi.table2$SIMPER<-round(fungi.table2$SIMPER,0)

# Merging everything into Table 2
fungi.table2$Taxon<-row.names(fungi.table2)
fungi.man.p.df$Taxon<-row.names(fungi.man.p.df)

#Final table
fungi.table2.final<-merge(fungi.table2,fungi.man.p.df,by="Taxon",all=TRUE)

# Round p-values to 2 decimals
fungi.table2.final$`P-Value`<-round(fungi.table2.final$`P-Value`,3)

# Put P-values in parantheses
fungi.table2.final$P2<-paste("(",fungi.table2.final$`P-Value`,")",sep="")

# Add + signs to Difference factors
for(i in 1:length(fungi.table2.final$Taxon)) {
  if (fungi.table2.final[i,4]>0) {
    fungi.table2.final[i,4]<-paste("+",fungi.table2.final[i,4],sep="")}
}

# Combine Difference with P-values in single column
fungi.table2.final$`Difference % (P-Value)`<-paste(fungi.table2.final$Difference,fungi.table2.final$P2,sep=" ")

# Remove unnecessary columns from final table
fungi.table2.final<-fungi.table2.final[,-c(4,6,7)]

fungi.table2.final<-fungi.table2.final[c("Taxon","Total","Active","Difference % (P-Value)","SIMPER")]

# Reorder Taxon by Total Rel Abund
fungi.table2.final$Taxon<-as.factor(fungi.table2.final$Taxon)
fungi.table2.final$Taxon<-reorder(fungi.table2.final$Taxon,c(4,3,7,1,6,9,2,5,8))
fungi.table2.final<-fungi.table2.final[order(fungi.table2.final$Taxon),]

# Name Taxon row name
row.names(fungi.table2.final)<-fungi.table2.final$Taxon

# Remove Taxon column
fungi.table2.final<-fungi.table2.final[,-1]

# Rename rows in the data frame to environmental variables
row.names(fungi.table2.final)<-c("Leotiomycetes","Sordariomycetes","Dothideomycetes","Agaricomycetes","Tremellomycetes","Microbotryomycetes","Eurotiomycetes","Unclassified","Other")

# Rename columns
colnames(fungi.table2.final)<-c("% Total","% Active", "% Difference (P-value)","% Dissimilarity Contribution")

# Set pander options (table formatting)
set.alignment('right',row.names = 'left')
pander(fungi.table2.final,caption = "Table 4. Relative abundance of total and active fungal taxa (mean ± standard deviation) with mean percent difference from total community.")
```

**Supplemental Table S4.** List of "Other" fungi (<1% relative abundance)
```{r}
fungi.other.names
```

###Supplemental Figure S2 - Percent Change of Bacterial and Fungal Taxa Relative Abundance from Total Community by Site
```{r Fig_S2_Bacteria}
###---------------- Initial BACTERIAL Site*Community setup

# Create new dataframe
bact.tax.sites<-bact.order.merge[, -c(14,15,28,29)]

# Assign row names as Taxon
row.names(bact.tax.sites)<-bact.tax.sites$Taxon

# Remove taxon column
bact.tax.sites<-bact.tax.sites[2:25]

# Convert to matrix
bact.tax.sites.final<-as.data.frame(t(as.matrix(bact.tax.sites)))

# Add in columns for Site and Community
bact.tax.sites.final$Site<-c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")
bact.tax.sites.final$Community<-c("Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active")

#Create matrix of numeric columns
bact.tax.sites.bind<-as.matrix(cbind(bact.tax.sites.final[1:13]))

bact.site.manova<-manova(bact.tax.sites.bind~Site*Community, data=bact.tax.sites.final)
bact.site.man.sum<-summary.aov(bact.site.manova)
bact.site.man.sum 

###-------------- Setting up plotting data

# Separate Total and Active Communities into dataframes
bact.tax.sites.total<-bact.tax.sites.final[1:12,]
bact.tax.sites.active<-bact.tax.sites.final[13:24,]

# Calculate Site Means for Total
bact.tax.sites.total.means<-aggregate(cbind(Acidobacteria_Gp1,Actinomycetales,Burkholderiales,Caulobacterales,Flavobacteriales,Other,Planctomycetales,Pseudomonadales,Rhizobiales,Rhodospirillales,Sphingobacteriales,Sphingomonadales,Unclassified)~Site+Community,bact.tax.sites.total,mean)

# Create new ID column with Site and Community combined for Active Bacteria
bact.tax.sites.active$ID<-paste(bact.tax.sites.active$Site,bact.tax.sites.active$Community,sep="_")

# Create new ID column with Site and Community combined for Total Means
bact.tax.sites.total.means$ID<-paste(bact.tax.sites.total.means$Site,bact.tax.sites.total.means$Community,sep="_")

# Adding "Total" prefix to column names
colnames(bact.tax.sites.total.means)<-paste("Total", colnames(bact.tax.sites.total.means),sep="_")
bact.tax.sites.total.means$Site<-bact.tax.sites.total.means$Total_Site
bact.tax.sites.total.means<-bact.tax.sites.total.means[c(1:15,17)]

# Merge Total and Active df's by ID
bact.tax.sites.merge<-merge(bact.tax.sites.active,bact.tax.sites.total.means,by="Site", all=TRUE)

# Make new column with Site_Plot_Community
bact.tax.sites.merge$ID2<-c("Active_A_1","Active_A_2","Active_A_3","Active_B_1","Active_B_2","Active_B_3","Active_C_1","Active_C_2","Active_C_3","Active_D_1","Active_D_2","Active_D_3")

# Make ID as rownames
row.names(bact.tax.sites.merge)<-bact.tax.sites.merge$ID2

# Split into two matrices of numeric values
# Active plot abundances
bact.tax.sites.active.abund<-as.matrix(bact.tax.sites.merge[2:14])

# Total mean abundace
bact.tax.sites.total.means.abund<-as.matrix(bact.tax.sites.merge[19:31])

# Calculate site differences
bact.tax.sites.diff<-as.data.frame(bact.tax.sites.active.abund-bact.tax.sites.total.means.abund)

# Add column for Site and Community
bact.tax.sites.diff$Site<-c("A","A","A","B","B","B","C","C","C","D","D","D")

# Aggregate by site mean
bact.tax.sites.diff.means<-aggregate(cbind(Acidobacteria_Gp1,Actinomycetales,Burkholderiales,Caulobacterales,Flavobacteriales,Other,Planctomycetales,Pseudomonadales,Rhizobiales,Rhodospirillales,Sphingobacteriales,Sphingomonadales,Unclassified)~Site,bact.tax.sites.diff,mean)

# Calculate sd
bact.tax.sites.diff.sd<-aggregate(cbind(Acidobacteria_Gp1,Actinomycetales,Burkholderiales,Caulobacterales,Flavobacteriales,Other,Planctomycetales,Pseudomonadales,Rhizobiales,Rhodospirillales,Sphingobacteriales,Sphingomonadales,Unclassified)~Site,bact.tax.sites.diff,sd)

# Melt Mean df
bact.mean.melt<-melt(bact.tax.sites.diff.means,id="Site")

# Rename columns
colnames(bact.mean.melt)<-c("Site","Taxon","Mean_Diff")

# Create Taxon vector
tax.order<-c(13,13,13,13,11,11,11,11,4,4,4,4,10,10,10,10,7,7,7,7,6,6,6,6,3,3,3,3,1,1,1,1,12,12,12,12,2,2,2,2,5,5,5,5,8,8,8,8,9,9,9,9)

bact.mean.melt$Taxon<-reorder(bact.mean.melt$Taxon,tax.order)

###-------------Build figure

# Initial ggplot figure
bact.tax.sites.diff.fig<-ggplot(bact.mean.melt,aes(x=Taxon,y=Mean_Diff,fill=Site))+geom_bar(width=0.5,colour="black",position=position_dodge(width = 0.5),stat="identity") + geom_hline(yintercept=0) + labs(x="",y="% Change from Total Bacterial Community") + theme(axis.ticks.y = element_blank(), axis.ticks.x = element_blank(), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.key = element_blank()) + scale_fill_manual("Site",labels=c("A","B","C","D"),values=c("white","light gray","dark gray","black")) + theme(axis.text.x=element_text(angle=90,vjust=1), panel.border=element_rect(colour = "black",fill=NA)) + theme(axis.text.x=element_text(size=12, color = "black"),axis.text.y=element_text(size = 12, color = "black"),axis.title.y=element_text(size = 12,color = "black"))+ annotate("text",x=12,y=5,label="*",size=12)
```

```{r Fig_S2_Fungi}
###--------------Initial FUNGAL Site*Community setup

# Create new dataframe
fungi.tax.sites<-fungi.merge[, -c(14,15,28,29)]

# Assign row names as Taxon
row.names(fungi.tax.sites)<-fungi.tax.sites$Taxon

# Remove taxon column
fungi.tax.sites<-fungi.tax.sites[2:25]

# Convert to matrix
fungi.tax.sites.final<-as.data.frame(t(as.matrix(fungi.tax.sites)))

# Add in columns for Site and Community
fungi.tax.sites.final$Site<-c("A","A","A","B","B","B","C","C","C","D","D","D","A","A","A","B","B","B","C","C","C","D","D","D")
fungi.tax.sites.final$Community<-c("Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active","Active")

#Create matrix of numeric columns
fungi.tax.sites.bind<-as.matrix(cbind(fungi.tax.sites.final[1:9]))

# MANOVA
fungi.site.manova<-manova(fungi.tax.sites.bind~Site*Community, data=fungi.tax.sites.final)
fungi.site.man.sum<-summary.aov(fungi.site.manova)
fungi.site.man.sum

###-------------- Setting up plotting data

# Separate Total and Active Communities into dataframes
fungi.tax.sites.total<-fungi.tax.sites.final[1:12,]
fungi.tax.sites.active<-fungi.tax.sites.final[13:24,]

# Calculate Site Means for Total
fungi.tax.sites.total.means<-aggregate(cbind(Agaricomycetes,Dothideomycetes,Eurotiomycetes,Leotiomycetes,Microbotryomycetes,Other,Sordariomycetes,Tremellomycetes,Unclassified)~Site+Community,fungi.tax.sites.total,mean)

# Create new ID column with Site and Community combined for Active Fungi
fungi.tax.sites.active$ID<-paste(fungi.tax.sites.active$Site,fungi.tax.sites.active$Community,sep="_")

# Create new ID column with Site and Community combined for Total Means
fungi.tax.sites.total.means$ID<-paste(fungi.tax.sites.total.means$Site,fungi.tax.sites.total.means$Community,sep="_")

# Adding "Total" prefix to column names
colnames(fungi.tax.sites.total.means)<-paste("Total", colnames(fungi.tax.sites.total.means),sep="_")
fungi.tax.sites.total.means$Site<-fungi.tax.sites.total.means$Total_Site
fungi.tax.sites.total.means<-fungi.tax.sites.total.means[c(1:11,13)]

# Merge Total and Active df's by ID
fungi.tax.sites.merge<-merge(fungi.tax.sites.active,fungi.tax.sites.total.means,by="Site", all=TRUE)

# Make new column with Site_Plot_Community
fungi.tax.sites.merge$ID2<-c("Active_A_1","Active_A_2","Active_A_3","Active_B_1","Active_B_2","Active_B_3","Active_C_1","Active_C_2","Active_C_3","Active_D_1","Active_D_2","Active_D_3")

# Make ID as rownames
row.names(fungi.tax.sites.merge)<-fungi.tax.sites.merge$ID2

# Split into two matrices of numeric values
# Active plot abundances
fungi.tax.sites.active.abund<-as.matrix(fungi.tax.sites.merge[2:10])

# Total mean abundace
fungi.tax.sites.total.means.abund<-as.matrix(fungi.tax.sites.merge[15:23])

# Calculate site differences
fungi.tax.sites.diff<-as.data.frame(fungi.tax.sites.active.abund-fungi.tax.sites.total.means.abund)

# Add column for Site
fungi.tax.sites.diff$Site<-c("A","A","A","B","B","B","C","C","C","D","D","D")

# Aggregate by site mean
fungi.tax.sites.diff.means<-aggregate(cbind(Agaricomycetes,Dothideomycetes,Eurotiomycetes,Leotiomycetes,Microbotryomycetes,Other,Sordariomycetes,Tremellomycetes,Unclassified)~Site,fungi.tax.sites.diff,mean)

# Calculate sd
fungi.tax.sites.diff.sd<-aggregate(cbind(Agaricomycetes,Dothideomycetes,Eurotiomycetes,Leotiomycetes,Microbotryomycetes,Other,Sordariomycetes,Tremellomycetes,Unclassified)~Site,fungi.tax.sites.diff,sd)

# Melt Mean df
fungi.mean.melt<-melt(fungi.tax.sites.diff.means,id="Site")

# Rename columns
colnames(fungi.mean.melt)<-c("Site","Taxon","Mean_Diff")

# Create Taxon vector
tax.fungi.order<-c(1,1,1,1,4,4,4,4,7,7,7,7,9,9,9,9,2,2,2,2,5,5,5,5,8,8,8,8,3,3,3,3,6,6,6,6)
fungi.mean.melt$Taxon<-reorder(fungi.mean.melt$Taxon,tax.fungi.order)

###-------------Build figure

# Initial ggplot figure
fungi.tax.sites.diff.fig<-ggplot(fungi.mean.melt,aes(x=Taxon,y=Mean_Diff,fill=Site))+geom_bar(width=0.5,colour="black",position=position_dodge(width = 0.5),stat="identity") + geom_hline(yintercept=0) + labs(x="",y="% Change from Total Fungal Community") + theme(axis.ticks.y = element_blank(), axis.ticks.x = element_blank(), panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), legend.key = element_blank()) + scale_fill_manual("Site",labels=c("A","B","C","D"),values=c("white","light gray","dark gray","black")) + theme(axis.text.x=element_text(angle=90,vjust=1), panel.border=element_rect(colour = "black",fill=NA)) + theme(axis.text.x=element_text(size=12, color = "black"),axis.text.y=element_text(size = 12, color = "black"),axis.title.y=element_text(size = 12,color = "black"))+ annotate("text",x=3,y=8,label="*",size=12) + annotate("text",x=9,y=8,label="*",size=12)
```

###Recreating Supplemental Figure S2.
```{r fig.width=10, fig.height=10}
grid.arrange(bact.tax.sites.diff.fig,fungi.tax.sites.diff.fig,ncol=1)
```
**Supplemental Figure S2.** Percent change in relative abundance from the Total community by Sites.

###Reproducibility
```{r}
# Provide the Session Information for reproducibility
devtools::session_info()
```
